src/osw/core.py

```diff
@@ -30,6 +30,31 @@
 from osw.wiki_tools import SearchParam
 from osw.wtsite import WtPage, WtSite
 
+# Reusable type definitions
+OVERWRITE_OPTIONS = Union[
+    bool,  # False: never overwrite a property,
+    # True: always overwrite a property
+    Literal[
+        "only empty",  # only overwrite if the property is empty
+        # "merge",  # todo: implement
+        # "append",  # don't replace the properties but for properties of type array or
+        # dict, append the values of the local entity to the remote entity, make sure
+        # to not append duplicates # todo: implement
+        # "only older",  # todo: implement read out from the version history of the page
+    ],
+]
+OVERWRITE_CLASS_OPTIONS = Union[
+    OVERWRITE_OPTIONS,  # All of the above which will serve as a fallback
+    Literal[
+        "replace remote",  # replace the entity with the new one, removes all
+        # properties that are not present in the local entity
+        "keep existing",  # keep the entity, does not add or remove any properties,
+        # if the page exists, the entity is not stored
+    ],
+    None,  # Not an option to choose from, will be replaced by the default
```

Comment:
'None' and 'false' should alse be enum values for consistancy and autocomplete support.

Reply:
Implemented with https://github.com/OpenSemanticLab/osw-python/commit/df2a469b850150d199e079609fff022080aa7195